service/start.go

```diff
@@ -26,18 +28,29 @@ func (service *Service) Start() (serviceIDs []string, err error) {
 		return
 	}
 	serviceIDs = make([]string, len(service.GetDependencies()))
+	var mutex sync.Mutex
 	i := 0
+	var wg sync.WaitGroup
 	for name, dependency := range service.GetDependencies() {
-		serviceIDs[i], err = dependency.Start(service, dependencyDetails{
+		d := dependencyDetails{
 			namespace:      service.namespace(),
 			dependencyName: name,
 			serviceName:    service.Name,
-		}, networkID)
-		i++
-		if err != nil {
-			break
 		}
+		wg.Add(1)
+		go func(service *Service, d dependencyDetails, name string, i int) {
+			serviceID, errStart := dependency.Start(service, d, networkID)
+			mutex.Lock()
```

Comment:
I'm not sure a mutex is needed here because in this code we have different goroutine that access different index so there is no reason to have conflicts but still is not something bad to do